File: exampleProjects/TypescriptExample/src/de/sots/cellarsandcentaurs/adapter/persistence/PersistedCreatures.ts
import { CreatureRepository } from './CreatureRepository';
import { CreatureEntity } from './CreatureEntity';
import { Creatures } from '../../domain/service/Creatures';
import { Creature } from '../../domain/model/Creature';
import { CreatureId } from '../../domain/model/CreatureId';
import { NoSuchCreatureException } from '../../domain/model/NoSuchCreatureException';
import { CreatureFacade } from "../../application";

export class PersistedCreatures implements Creatures {
    constructor(
        private repository: CreatureRepository
    ) {}

    async save(creature: Creature): Promise<void> {
        await this.repository.save(new CreatureEntity(creature.getId().id));
    }

    async find(id: CreatureId): Promise<Creature> {
        const creatureEntity = await this.repository.findOne(id.id);
        if (!creatureEntity) {
            throw new NoSuchCreatureException(id);
        }
        return new Creature(new CreatureId(creatureEntity.id), CreatureFacade.STANDARD_CREATURE_TYPE);
    }
}

File: exampleProjects/PythonExample/src/de/sots/cellarsandcentaurs/adapter/persistence/persisted_creatures.py
from de.sots.cellarsandcentaurs.domain.models import Creature, CreatureId as CrId, NoSuchCreatureException

from de.sots.cellarsandcentaurs.domain.services.creatures import Creatures as Crs

from ..persistence.creature_entity import CreatureEntity

class PersistedCreatures(Crs):
    def __init__(self, repository):
        self.repository = repository

    def save(self, creature: Creature):
        # Assuming CreatureEntity is the persistence equivalent within the system
        entity = CreatureEntity(creature.id.id)
        self.repository.save(entity)

    def find(self, creature_id: CrId) -> Creature:
        entity = next(
            (entity for entity in self.repository.find_by_id(creature_id.id)
             if entity.get_id() == creature_id.id), None
        )

        if entity is None:
            raise NoSuchCreatureException(creature_id)

        return Creature(CrId(entity.get_id()))